File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

/**
 * caesar cipher crypto program
 * 
 */

int k = 0;
char Encrypt(char a);
int GetK(int n);

int main(int argc, string argv[])
{
    if (argc !=2)
    {
        printf("Please use only one and positive integer\n");
        return 1;
    }
    k = GetK(atoi(argv[1]));
       
    // get the phrase from user
    string phrase = GetString();
        
    //encrypt user input
    for (int i = 0, n = strlen(phrase); i < n; i++)
    {
        Encrypt(phrase[i]);
        printf("%c", Encrypt(phrase[i]));
    }
    printf("\n");
    return 0;
}


//mod26 function to calculate the key
int GetK(int n)
{
    while (n - 26 >= 0)
        {
            n = n - 26;
        }
    return n;
}


//Encrypt a character
char Encrypt(char a)
{
    
    if (isalpha(a) != 0)
    {
        int j;
        j = (int) a;
        
        //encrypt a lower character
        if (islower(a) != 0)
        {
            j = j - 97;
            j = (j + k) % 26;
            j = j + 97;
        }
        else
        //encrypt a uppper character    
            if (isupper(a) != 0)
            {
                j = j - 65;
                j = (j + k) % 26;
                j = j + 65;
            }    
        a = (char) j; 
    }
    return a;
}

File: pset6/caesar.py
#include <stdio.h>
import cs50
import sys
import string

k = 0
newphrase = []

# main program
def main():
    # check that the client inputs are correct
    if len(sys.argv) != 2:
        print("Please use only one and positive integer")

    # mod26 to the k integer given by the client

    global k
    k = get_k(int(sys.argv[1]))

    # get the phrase from user
    phrase = cs50.get_string()

    # defining the new phrase as a list of phrase characters
    for i in range(len(phrase)):
        keychar = Encrypt(phrase[i])
        global newphrase
        newphrase.append(keychar)

    # print the new list of character
    for i in range(len(newphrase)):
        sys.stdout.write("{}".format(newphrase[i]))

    #end of printing!
    print("")

# gives the k script with mod26, usefull for the alphabet !
def get_k(n):
    n = n % 26
    return n

# encrypt a character
def Encrypt(a):
    if str.isalpha(a) == True:
        j = ord(a)

        # encrypt a lower character with mod26
        if str.islower(a) == True:
            j = j - 97
            j = (j + k) % 26
            j = j + 97

        # encrypt a uppper character with mod26
        else:
            if str.isupper(a) == True:
                j = j - 65
                j = (j + k) % 26
                j = j + 65

        # convert the ascii number in a character
        a = chr(j)
    return a

if __name__ == "__main__":
    main()
